packages/grafana-ui/src/components/Monaco/CodeEditor.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { withTheme } from '../../themes';
+import { Themeable } from '../../types';
+import { MonacoEditorProps } from 'react-monaco-editor';
+import { KeyCode, editor, KeyMod } from 'monaco-editor/esm/vs/editor/editor.api';
+import { useAsyncDependency } from '../../utils/useAsyncDependency';
+import { ErrorWithStack, LoadingPlaceholder } from '..';
+
+export type CodeEditorChangeHandler = (value: string) => void;
+
+const MonacoEditor: React.FC<MonacoEditorProps> = props => {
+  const { loading, error, dependency } = useAsyncDependency(
+    import(/* webpackChunkName: "react-monaco-editor" */ 'react-monaco-editor')
+  );
+
+  if (loading) {
+    return <LoadingPlaceholder text={'Loading...'} />;
+  }
+
+  if (error) {
+    return (
+      <ErrorWithStack
+        title="Code editor failed to load"
+        error={error}
+        errorInfo={{ componentStack: error?.stack || '' }}
+      />
+    );
+  }
+
+  const ReactMonaco = dependency.default;
+  return <ReactMonaco {...props} />;
+};
+
+interface CodeEditorProps extends Themeable {
+  value: string;
+  language: string;
+  width?: number | string;
+  height?: number | string;
+
+  readOnly?: boolean;
+  showMiniMap?: boolean;
+
+  /**
+   * Callback after the editor has mounted that gives you raw access to monaco
+   *
+   * @experimental
+   */
+  onEditorDidMount?: (editor: editor.IStandaloneCodeEditor) => void;
+
+  /** Handler to be performed when editor is blurred */
+  onBlur?: CodeEditorChangeHandler;
+
+  /** Handler to be performed when Cmd/Ctrl+S is pressed */
+  onSave?: CodeEditorChangeHandler;
+}
+
+class UnthemedCodeEditor extends React.PureComponent<CodeEditorProps> {
+  constructor(props: CodeEditorProps) {
+    super(props);
+  }
+
+  getEditorValue = () => '';
```

Comment:
Should this return the actual value?

Reply:
This function is re-assigned after monaco is instantiated. See `editorDidMount`